UC-Berkeley-Extension/global
# Pantheon deploy

## Asana Ticket(s)
- See ticket [Pantheon configuration testing](https://app.asana.com/0/1174970560648905/1194174850852899/f).

## Description

Enables a build of the site to be pushed to Pantheon as a deployment.

## Testing instructions
1. Checkout this branch
1. Run `composer install` from the project root.
2. Open the README and read the `Deploying to Pantheon` section.
3. Setup `terminus`, `drush` and `ssh` as described in the documentation
4. Build an artifact according to the documentation.
5. Go to https://deploy-berkeleyglobal.pantheonsite.io/ and confirm that the site functions.
6. Go to the Pantheon dashboard DEV site and merge the `deploy` environment
7. Go to https://dev-berkeleyglobal.pantheonsite.io/ and confirm that the site functions.

## Advanced testing

1. After doing the above, run `phing artifact -Dartifact.git.remote_branch=mytest`
1. Approve the deployment
1. Go to the Pantheon Multidev dashboard and Create an environment fo `mytest`
1. Go to https://mytest-berkeleyglobal.pantheonsite.io/ and confirm that the site functions.

## Remaining tasks
- [x] Confirm workflow needs for deployments
- [x] Add HTAUTH security to the Pantheon sites
- [ ] Upgrade the sandbox site
